polygeist/tools/cgeist/Lib/CodeGenTypes.cc

```diff
@@ -1906,5 +1906,18 @@ bool CodeGenTypes::getCPUAndFeaturesAttributes(
   return AddedAttr;
 }
 
+QualType CodeGenTypes::getPromotionType(QualType Ty) const {
+  if (CGM.getTarget().shouldEmitFloat16WithExcessPrecision()) {
+    if (Ty->isAnyComplexType()) {
+      QualType ElementType = Ty->castAs<clang::ComplexType>()->getElementType();
+      if (ElementType->isFloat16Type())
+        return CGM.getContext().getComplexType(CGM.getContext().FloatTy);
+    }
+    if (Ty->isFloat16Type())
+      return CGM.getContext().FloatTy;
```

Comment:
``` 
if (Ty->isRealFloatingType() && Ty->isFloat16Type())
  return CGM.getContext().FloatTy;

Reply:
I copied this one from `CGExprScalar.cpp`